scripts/import_proxy_certs.sh

```diff
@@ -0,0 +1,67 @@
+#/!bin/sh
+#
+# Copyright (C) 2020 Bosch.IO GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# License-Filename: LICENSE
+
+if [ -z "$https_proxy" ]; then
+    echo "Skipping the import of certificates as no HTTPS proxy is set."
+    exit
+fi
+
+CONNECT_SERVER="jcenter.bintray.com:443"
+
+FILE="proxy.crt"
+FILE_PREFIX="proxy-"
+
+REGEX_BEGIN="/^-----BEGIN CERTIFICATE-----$/"
+REGEX_END="/^-----END CERTIFICATE-----$"
+
+# Pick a server to connect to that is used during the Gradle build, and which reports the proxy's certificate instead of
+# its own.
+echo "Getting the proxy's certificates..."
+openssl s_client -showcerts -proxy ${https_proxy#*//} -connect $CONNECT_SERVER | \
+    sed -n "$REGEX_BEGIN,$REGEX_END/p" > $FILE
+
+# Split the potentially multiple certifcates into multiple files to avoid only the first certificate being imported.
+echo "Splitting proxy certificates to separate files..."
+csplit -f $FILE_PREFIX -b "%02d.crt" -z $FILE "$REGEX_BEGIN" "{*}"
+
+KEYTOOL=$(realpath $(command -v keytool))
+
+for KEYSTORE_CANDIDATE in "$(realpath -m $(dirname $KEYTOOL)/../lib/security/cacerts)" "$(realpath -m $(dirname $KEYTOOL)/../jre/lib/security/cacerts)"; do
```

Comment:
> You could put `$(realpath -m $(dirname $KEYTOOL))` to a variable to avoid repeating it here.

Not really, as one of the points of `realpath` is to resolve the `..` that occurs in the constructed paths. But I've omitted the `realpath` when assigning `KEYTOOL` instead, that was not strictly necessary.

Reply:
Turns out the `realpath` in `KEYTOOL=$(realpath $(command -v keytool))` *is* needed. I simply tried it and the import then fails (without me looking deeper into the reason, TBH).